azure-quantum/azure/quantum/storage.py

```diff
@@ -74,24 +68,57 @@ def upload_blob(
     blob.upload_blob(data, content_settings=content_settings)
     logger.debug(f"  - blob '{blob_name}' uploaded. generating sas token.")
 
-    if return_sas_token:
-        sas_token = generate_blob_sas(
-            blob.account_name,
-            blob.container_name,
-            blob.blob_name,
-            account_key=blob.credential.account_key,
-            permission=BlobSasPermissions(read=True),
-            expiry=datetime.utcnow() + timedelta(days=14)
-        )
+    if returnSasToken:
+        uri = get_blob_uri_with_sas_token(blob)
+    else:
+        uri = remove_sas_token(blob.url)
 
-        uri = blob.url + "?" + sas_token
+    logger.debug(f"  - blob access url: '{uri}'.")
+
+    return uri
+
+def append_blob(container: ContainerClient, blob_name: str, content_type:str, content_encoding:str, data: Any, returnSasToken: bool=True, metadata: Dict[str, str] = None) -> str:
```

Comment:
This line is a bit long, to shorten it I'd suggest adding newlines after each comma